tests/src/Simple/PInvoke/PInvokeNative.vcxproj

```diff
@@ -0,0 +1,58 @@
+<Project ToolsVersion="14.0"  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+ <ItemGroup>
+    <ClCompile Include="PInvokeNative.cpp" />
+ </ItemGroup>
+  
+ <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>    
+    <OutputPath>$(MSBuildProjectDirectory)\bin\$(Configuration)\native\</OutputPath>
+    <IntermediateOutputPath>$(MSBuildProjectDirectory)\obj\$(Configuration)\native\</IntermediateOutputPath>
+    <TargetOS Condition="'$(TargetOS)' == '' and '$(OS)' == 'Unix' and Exists('/Applications')">OSX</TargetOS>
+    <TargetOS Condition="'$(TargetOS)' == ''">$(OS)</TargetOS>
+    <OutputName>PInvokeNative</OutputName>
+    <NativeObjectExt Condition="'$(TargetOS)' == 'Windows_NT'">.obj</NativeObjectExt>
+    <NativeObjectExt Condition="'$(TargetOS)' != 'Windows_NT'">.o</NativeObjectExt>
+    <PInvokeNativeObject>$(IntermediateOutputPath)$(OutputName)$(NativeObjectExt)</PInvokeNativeObject>
+    <PInvokeNativeBinary>$(OutputPath)$(OutputName).dll</PInvokeNativeBinary>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)\..\..\..\..\src\BuildIntegration\Microsoft.NETCore.Native.Windows.props" Condition="'$(TargetOS)' == 'Windows_NT'" />
+  <Import Project="$(MSBuildThisFileDirectory)\..\..\..\..\src\BuildIntegration\Microsoft.NETCore.Native.Unix.props" Condition="'$(TargetOS)' != 'Windows_NT'" />
+
+  <Target Name="Build" DependsOnTargets="Compile;Link"> </Target>
+  <Target Name="Compile"
+      Inputs="@(ClCompile)"
+      Outputs="$(PInvokeNativeObject)">
+
+    <ItemGroup>
+      <CompilerArg Include="@(ClCompile)" />
+      <CompilerArg Include="/Fo$(PInvokeNativeObject)" Condition="'$(OS)' == 'Windows_NT'" />
+      <CompilerArg Include="-o $(PInvokeNativeObject)" Condition="'$(OS)' != 'Windows_NT'" />
+      <CompilerArg Include="@(CppCompilerAndLinkerArg)" />
+    </ItemGroup>
+
+    <MakeDir Directories="$(IntermediateOutputPath)" />
+    <Exec Command="$(CppCompiler) @(CompilerArg, ' ')" Condition="'$(OS)' != 'Windows_NT'" />
+    <WriteLinesToFile File="$(IntermediateOutputPath)cl.rsp" Lines="@(CompilerArg)" Overwrite="true"  Condition="'$(OS)' == 'Windows_NT'"/>
+    <Exec Command="$(CppCompiler) @&quot;$(IntermediateOutputPath)cl.rsp&quot;" Condition="'$(OS)' == 'Windows_NT'" />
+  </Target>
+
+  <Target Name="Link"
+      Inputs="$(PInvokeNativeObject)"
+      Outputs="$(PInvokeNativeBinary)"
+      DependsOnTargets="Compile">
+
+    <ItemGroup>
+      <CustomLinkerArg Include="$(PInvokeNativeObject)" />
```

Comment:
I do not think you need to replicate the Link step here. You just need to add the .obj file for your C++ file to the .objs passed to the linker.